### PR TITLE
Ground work for Python 3

### PR DESCRIPTION
**Pull request by Cadair**
This adds a copy of Six to the source tree and uses it for a few imports.

This is a reduced version of #1449, without the modernize runs, which will follow in seperate PRs module by module.

### CODE REVIEW

File: requirements/extra.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ pytest-cov
 sqlalchemy
 scikit-image
 glymur
-suds
+suds-juko
```
Comment (ayshih) [inline requirements/extra.txt]: typo?

File: sunpy/extern/setup_package.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import
+
+import os
+
+
+def get_package_data():
+    paths = [os.path.join('js', '*.js')]
+    return {'astropy.extern': paths}
```
Comment (ayshih) [inline sunpy/extern/setup_package.py]: astropy?

File: sunpy/extern/six.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            # Attribute error if the six module isn't what it should be and
+            # doesn't have a .__version__; ValueError if the version string
+            # exists but is somehow bogus/unparseable
+            continue
+    else:
+        raise ImportError(
+            "SunPy requires the 'six' module of minimum version {0}; "
+            "normally this is bundled with the astropy package so if you get "
+            "this warning consult the packager of your SunPy "
+            "distribution.".format(_SIX_MIN_VERSION))
```
Comment (ayshih) [inline sunpy/extern/six.py]: "bundled with astropy"?

File: requirements/extra.txt (hunk 1 above)
Comment (Cadair) [inline requirements/extra.txt]: nope

**Comment by ayshih** [comment]
:+1:

**Comment by dpshelio** [comment]
:+1: from me
But can I  get a explanation of why we need `six` inside sunpy? is it a modified version of it?  cannot we just import it?

**Comment by Cadair** [comment]
@dpshelio see: https://github.com/sunpy/sunpy/pull/1449#issuecomment-112490697

**Comment by Cadair** [comment]
@nabobalis we are go.

**Comment by nabobalis** [comment]
@Cadair TIME TILL DROP 30 seconds.

**Pull request merged**